### PR TITLE
Update poi to v12

### PR DESCRIPTION
**Pull request by borna223**
Updated poi to v12.

### CODE REVIEW

File: poi.config.js [hunk 1]
```diff
@@ -1,51 +1,53 @@
 require('dotenv').config();
+const { BundleAnalyzerPlugin } = require('webpack-bundle-analyzer');
 const config = require('./server/config');
 const path = require('path');
 
 const isProduction = process.env.NODE_ENV === 'production';
-const aliases = {
-  '@': path.resolve(__dirname, './client')
-};
+const aliases = { '@': path.resolve(__dirname, './client') };
```
Comment (vladimyr) [inline poi.config.js]: Don't inline this, we might have larger map of those in future.

File: poi.config.js [hunk 2]
```diff
@@ -1,51 +1,53 @@
...
+const extensions = ['.vue'];
 
 const devServer = {
-  headers: {
-    'X-Powered-By': 'Webpack DevSever'
-  },
```
Comment (vladimyr) [inline poi.config.js]: Why did you remove this?

File: poi.config.js [hunk 3]
```diff
@@ -1,51 +1,53 @@
...
-    filename: 'admin/index.html',
-    excludeChunks: ['student']
-  }, {
-    filename: 'index.html',
-    excludeChunks: ['admin']
-  }],
   chainWebpack(config) {
     configureModuleResolution(config);
     config.resolve.alias.merge(aliases);
+    config.resolve.extensions.merge(extensions);
```
Comment (vladimyr) [inline poi.config.js:49]: So this is now required? 🤔

**Review by vladimyr — CHANGES_REQUESTED** [review]

File: poi.config.js (hunk 3 above)
Comment (borna223) [reply poi.config.js:49]: https://github.com/egoist/poi/pull/528

File: poi.config.js [hunk 4]
```diff
@@ -15,37 +17,42 @@ const devServer = {
...
+  hot: true,
+  hotEntries: ['admin', 'student']
 };
 
-module.exports = args => ({
+module.exports = {
   plugins: [
-    require('@poi/plugin-eslint')({ command: '*' }),
-    require('@poi/plugin-bundle-report')()
+    '@poi/eslint'
```
Comment (vladimyr) [inline poi.config.js:30]: @borna223 This would work both in `dev` and `prod` mode? Just double checkin...

File: poi.config.js [hunk 5]
```diff
@@ -15,37 +17,42 @@ const devServer = {
...
-    filename: 'index.html',
-    excludeChunks: ['admin']
-  }],
   chainWebpack(config) {
     configureModuleResolution(config);
     config.resolve.alias.merge(aliases);
+    config.resolve.extensions.merge(extensions);
+  },
+  configureWebpack(config) {
+    if (!process.argv.includes('--bundle-report')) return;
```
Comment (vladimyr) [inline poi.config.js]: Um, don't do this. Use [`minimist`](https://npm.im/minimist) instead and check if flag is specified.

**Review by vladimyr — CHANGES_REQUESTED** [review]

**Pull request merged**